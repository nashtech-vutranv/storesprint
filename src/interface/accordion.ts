type AccordionName =
  | 'warehouse'
  | 'product'
  | 'relationship'
  | 'assignedProduct'
  | 'user'
  | 'listingStatus'
  | 'mappings'
  | 'defaultPropertyValue'
  | 'order'
  | 'productCategoryMappings'
  | 'productPropertyMappings'
  | 'courierMappings'
  | 'deliveryTypeMappings'

interface IAccordion extends Record<AccordionName, string> {}

export type {AccordionName, IAccordion}
