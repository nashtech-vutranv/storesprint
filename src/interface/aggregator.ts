interface IAggregator {
    id: string;
    name: string;
    version: number;
    status: string;
    createdAt?: string;
    modifiedAt?: string;
}

export type {IAggregator}