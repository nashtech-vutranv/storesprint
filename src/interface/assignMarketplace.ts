interface IAssignMarketplace {
  organizationId: string
  siteId: string
  siteLocaleId: string
  marketplaceIds: string[]
  isAllProducts: boolean
}

interface IMarketplacesDialog {
  id: string
  version: number
  name: string
  assignedStatus: 'ASSIGNED' | 'UNASSIGNED' | 'BOTH' | null,
  createdAt?: string
  modifiedAt?: string
}

interface IRequestBodyCalculateAssignMarketplaces extends IAssignMarketplace {
  isUnassignedMarketplace: boolean
  productIds: string[]
}

export type {IAssignMarketplace, IMarketplacesDialog, IRequestBodyCalculateAssignMarketplaces}