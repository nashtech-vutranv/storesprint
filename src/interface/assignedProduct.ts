interface IAssignedMarketplaces {
  organizationIds: string[]
  siteIds: string[]
  siteLocaleIds: string[]
}

interface IBodyRequestAssignMarketplace {
  id: string
  version: number
  name: string
  assignedStatus: 'ASSIGNED' | 'UNASSIGNED' | 'BOTH'
}

interface IMarketPlacesAssignedProduct {
  id: string
  version: number
  erpId: string
  name: string
  isMarketplaceSynced?: boolean
}

interface IBodyRequestAssignedProducts {
  organizationId: string
  siteId: string
  siteLocaleId: string
  marketplaceIds: string[]
  search: string
  isAllMarketplaces: boolean
  isUnassignedMarketplaces: boolean
}

interface IBodyRequestAssignProductsToMarketplaces {
  organizationId: string
  siteId: string
  siteLocaleId: string
  productIds: string[]
  isAllProducts: boolean
  assignedMarketplace: IBodyRequestAssignMarketplace[]
}

interface IAssignedProducts {
  erpId: string
  id: string
  name: string
  createdAt: string
  modifiedAt: string
  version: number
  status: string
  marketplaces: IMarketPlacesAssignedProduct[]
}

export type {
  IAssignedMarketplaces,
  IBodyRequestAssignedProducts,
  IAssignedProducts,
  IBodyRequestAssignProductsToMarketplaces,
}