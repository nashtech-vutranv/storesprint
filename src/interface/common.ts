interface IBaseEntities {
  id: string
  erpId: string
  createdAt: string
  modifiedAt: string
  name: string
  version: number
}

interface IBaseEntitiesOmitErpId extends Omit<IBaseEntities, 'erpId'> {}

interface IBaseEntitiesOmitName extends Omit<IBaseEntities, 'name'> {}

interface IBaseEntitiesOmitNameAndErpId
  extends Omit<IBaseEntities, 'erpId' | 'name'> {}

export type {
  IBaseEntities,
  IBaseEntitiesOmitErpId,
  IBaseEntitiesOmitName,
  IBaseEntitiesOmitNameAndErpId,
}
