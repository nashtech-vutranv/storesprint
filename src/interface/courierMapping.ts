interface ICreateUpdateCourierMapping {
  id: string
  mmsValue: string
  marketplaceTypeId: string
  marketplaceValue: string
  marketplaceCode: string
  version: number
  status: string
}

export type {
  ICreateUpdateCourierMapping
}
