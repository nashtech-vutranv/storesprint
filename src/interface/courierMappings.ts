import {IBaseEntitiesOmitNameAndErpId} from '.'

interface ICourierMapping extends IBaseEntitiesOmitNameAndErpId {
    marketplaceTypeId: string
    mappingTypeId: string
    mmsValue: string
    marketplaceValue: string
    marketplaceType: {
        id: string
        name: string
    }
    mappingType: {
        id: string
        name: string
    }
    mappingTarget: {
        id: string
        marketplaceTypeId: string
        mappingTypeId: string
        marketplaceCode: string
        marketplaceValue: string
    }
}

export type {
    ICourierMapping
}