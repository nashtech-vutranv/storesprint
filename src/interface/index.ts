export * from './common'
export * from './accordion'
export * from './aggregator'
export * from './assignMarketplace'
export * from './assignedProduct'
export * from './currency'
export * from './customers'
export * from './defaultPropertyValue'
export * from './filters'
export * from './integration'
export * from './language'
export * from './listingState'
export * from './location'
export * from './mappings'
export * from './marketplace'
export * from './marketplaceRelationship'
export * from './menu'
export * from './order'
export * from './organization'
export * from './pageInfo'
export * from './pagination'
export * from './productMaketplace'
export * from './productProperties'
export * from './products'
export * from './relationshipAM'
export * from './resource'
export * from './response'
export * from './selectOption'
export * from './shopify'
export * from './tiktok'
export * from './topbar'
export * from './userSetting'
export * from './warehouse'
export * from './productPropertyMappings'
export * from './productCategoryMappings'
export * from './mappingValue'
export * from './courierMappings'
export * from './importProductProperties'
export * from './params'
export * from './returnRefund'
export * from './marketplaceIntegration'
export * from './delivery'
export * from './ups'
export * from './user'
export * from './previousPage'