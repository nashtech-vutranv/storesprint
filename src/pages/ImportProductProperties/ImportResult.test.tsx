import {render} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import {IImportProductPropertiesResult} from '../../interface'
import {ImportResult} from './'

jest.mock('react-router-dom', () => {
  return {
    ...jest.requireActual('react-router-dom'),
    useParams: jest.fn(),
  }
})

beforeEach(() => {
  window.history.pushState(
    {
      errorMessage: '',
      invalidProperties: [],
      success: 0,
      total: 0,
    } as IImportProductPropertiesResult,
    '',
    ''
  )
})

describe('Import result Page', () => {
  it('should match snapshot', async () => {
    const importResultTree = render(
      <MemoryRouter>
        <ImportResult />
      </MemoryRouter>
    )
    expect(importResultTree).toMatchSnapshot()
  })
})
