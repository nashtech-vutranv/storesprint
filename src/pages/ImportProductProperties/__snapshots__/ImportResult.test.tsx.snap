// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import result Page should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-form mt-3 card"
      >
        <div
          class="card-header"
        >
          <h4
            class="card-form__title"
          >
            import_product_properties_result_title
          </h4>
        </div>
        <div
          class="card-body"
        >
          <div>
            <p>
              import_product_properties_result_insert_succesfully
            </p>
            <p>
              import_product_properties_result_invalid_properties
            </p>
          </div>
          <div
            class="mt-2"
          >
            <div
              class="p-datatable p-component p-datatable-responsive-scroll data-table-mh"
              data-scrollselectors=".p-datatable-wrapper"
              pr_id_1=""
            >
              <div
                class="p-datatable-wrapper"
              >
                <table
                  class="p-datatable-table"
                  role="table"
                >
                  <thead
                    class="p-datatable-thead"
                  >
                    <tr
                      role="row"
                    >
                      <th
                        class=""
                        role="columnheader"
                      >
                        <div
                          class="p-column-header-content"
                        >
                          <span
                            class="p-column-title"
                          >
                            import_product_properties_result_field_row_number
                          </span>
                        </div>
                      </th>
                      <th
                        class=""
                        role="columnheader"
                      >
                        <div
                          class="p-column-header-content"
                        >
                          <span
                            class="p-column-title"
                          >
                            import_product_properties_result_erpId
                          </span>
                        </div>
                      </th>
                      <th
                        class=""
                        role="columnheader"
                      >
                        <div
                          class="p-column-header-content"
                        >
                          <span
                            class="p-column-title"
                          >
                            import_product_properties_result_name
                          </span>
                        </div>
                      </th>
                      <th
                        class=""
                        role="columnheader"
                      >
                        <div
                          class="p-column-header-content"
                        >
                          <span
                            class="p-column-title"
                          >
                            import_product_properties_result_data_type
                          </span>
                        </div>
                      </th>
                      <th
                        class=""
                        role="columnheader"
                      >
                        <div
                          class="p-column-header-content"
                        >
                          <span
                            class="p-column-title"
                          >
                            import_product_properties_result_is_locale_sensitive
                          </span>
                        </div>
                      </th>
                      <th
                        class=""
                        role="columnheader"
                      >
                        <div
                          class="p-column-header-content"
                        >
                          <span
                            class="p-column-title"
                          >
                            import_product_properties_result_status
                          </span>
                        </div>
                      </th>
                      <th
                        class=""
                        role="columnheader"
                      >
                        <div
                          class="p-column-header-content"
                        >
                          <span
                            class="p-column-title"
                          >
                            import_product_properties_result_import_failed_reason
                          </span>
                        </div>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="p-datatable-tbody"
                  >
                    <tr
                      class="p-datatable-emptymessage"
                      role="row"
                    >
                      <td
                        colspan="7"
                        role="cell"
                      >
                        import_product_properties_result_empty_message
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-center mt-2"
          >
            <button
              class="me-2 btn btn-success"
              type="button"
            >
              common_ok
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-form mt-3 card"
    >
      <div
        class="card-header"
      >
        <h4
          class="card-form__title"
        >
          import_product_properties_result_title
        </h4>
      </div>
      <div
        class="card-body"
      >
        <div>
          <p>
            import_product_properties_result_insert_succesfully
          </p>
          <p>
            import_product_properties_result_invalid_properties
          </p>
        </div>
        <div
          class="mt-2"
        >
          <div
            class="p-datatable p-component p-datatable-responsive-scroll data-table-mh"
            data-scrollselectors=".p-datatable-wrapper"
            pr_id_1=""
          >
            <div
              class="p-datatable-wrapper"
            >
              <table
                class="p-datatable-table"
                role="table"
              >
                <thead
                  class="p-datatable-thead"
                >
                  <tr
                    role="row"
                  >
                    <th
                      class=""
                      role="columnheader"
                    >
                      <div
                        class="p-column-header-content"
                      >
                        <span
                          class="p-column-title"
                        >
                          import_product_properties_result_field_row_number
                        </span>
                      </div>
                    </th>
                    <th
                      class=""
                      role="columnheader"
                    >
                      <div
                        class="p-column-header-content"
                      >
                        <span
                          class="p-column-title"
                        >
                          import_product_properties_result_erpId
                        </span>
                      </div>
                    </th>
                    <th
                      class=""
                      role="columnheader"
                    >
                      <div
                        class="p-column-header-content"
                      >
                        <span
                          class="p-column-title"
                        >
                          import_product_properties_result_name
                        </span>
                      </div>
                    </th>
                    <th
                      class=""
                      role="columnheader"
                    >
                      <div
                        class="p-column-header-content"
                      >
                        <span
                          class="p-column-title"
                        >
                          import_product_properties_result_data_type
                        </span>
                      </div>
                    </th>
                    <th
                      class=""
                      role="columnheader"
                    >
                      <div
                        class="p-column-header-content"
                      >
                        <span
                          class="p-column-title"
                        >
                          import_product_properties_result_is_locale_sensitive
                        </span>
                      </div>
                    </th>
                    <th
                      class=""
                      role="columnheader"
                    >
                      <div
                        class="p-column-header-content"
                      >
                        <span
                          class="p-column-title"
                        >
                          import_product_properties_result_status
                        </span>
                      </div>
                    </th>
                    <th
                      class=""
                      role="columnheader"
                    >
                      <div
                        class="p-column-header-content"
                      >
                        <span
                          class="p-column-title"
                        >
                          import_product_properties_result_import_failed_reason
                        </span>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="p-datatable-tbody"
                >
                  <tr
                    class="p-datatable-emptymessage"
                    role="row"
                  >
                    <td
                      colspan="7"
                      role="cell"
                    >
                      import_product_properties_result_empty_message
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          class="d-flex justify-content-center mt-2"
        >
          <button
            class="me-2 btn btn-success"
            type="button"
          >
            common_ok
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
